lib/models/project.js

```diff
@@ -408,16 +435,83 @@ class Project {
   }
 
   /**
-    Discovers all addons for this project and stores their names and
-    package.json contents in this.addonPackages as key-value pairs
-
-    @private
-    @method discoverAddons
+   * Discovers all addons for this project and stores their names and
+   * package.json contents in this.addonPackages as key-value pairs.
+   *
+   * Any packageInfos that we find that are marked as not valid are excluded.
+   *
+   * @private
+   * @method discoverAddons
    */
   discoverAddons() {
-    let addonsList = this.addonDiscovery.discoverProjectAddons(this);
+    let pkgInfo = this.packageInfoCache.getEntry(this.root);
+
+    let addonPackageList = [];
+
+    if (this.isEmberCLIAddon()) {
+      addonPackageList.push(pkgInfo);
+    }
+
+    let cliInRepoAddons = pkgInfo.cliEntry
+      ? pkgInfo.cliEntry.inRepoAddons
+      : null;
+    if (cliInRepoAddons) {
+      addonPackageList = addonPackageList.concat(cliInRepoAddons);
+    }
+
+    if (pkgInfo.internalAddons) {
+      addonPackageList = addonPackageList.concat(pkgInfo.internalAddons);
+    }
+
+    // The 'dependencyPackages' object contains ALL dependencies, not just those
```

Comment:
The next 40-ish lines are very similar to the same code in discoverAddons() in lib/models/addon - this should be refactored to remove the duplication